aws-analytics-pinpoint/src/androidTest/java/com/amplifyframework/analytics/pinpoint/AnalyticsPinpointInstrumentedTest.java

```diff
@@ -57,8 +65,19 @@ public static void setUp() throws AmplifyException {
         Context context = ApplicationProvider.getApplicationContext();
         AmplifyConfiguration configuration = new AmplifyConfiguration();
         configuration.populateFromConfigFile(context, R.raw.amplifyconfiguration);
-        Amplify.addPlugin(new AmazonPinpointAnalyticsPlugin());
+        plugin = new AmazonPinpointAnalyticsPlugin();
+        Amplify.addPlugin(plugin);
         Amplify.configure(configuration, context);
+        analyticsClient = plugin.getAnalyticsClient();
```

Comment:
A better test pattern is to allow "injection" of a class' dependency via its constructor. So,

Instead of doing:
```
final class Container {
    private final Inside inside;
    ...
    Container() {
        this.inside = new DefaultInsideImpl();
    }

    @VisibleForTesting
    Inside getInside() {
        return inside;
    }
    ...
}
```
See if you can make this look more like:
```
final class Container {
    private final Inside inside;

    Container() {
        this(new DefaultInsideImpl());
    }

    @VisibleForTesting
    Container(Inside inside) {
        this.inside = inside;
    }
    ...
}
```

Then in your test, you'd have like:
```
@Before
public void setup() {
    this.analyticsClient = // Some method of obtaining one -- I forget how this works, its a factory right?
    this.plugin = new AmazonPinpointAnalyticsPlugin(analyticsClient);
}
```

Reply:
analyticsClient is created only when the Analytics is configured. So I don't see a way to obtain analyticsClient before the Analytics plugin is configure other than may be to recreate the configuration steps of Analytics inside the the test setup which feels redundant. Will discuss offline.